src/main/java/roomescape/controller/api/member/LoginApiController.java

```diff
@@ -0,0 +1,54 @@
+package roomescape.controller.api.member;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.validation.Valid;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import roomescape.controller.api.member.dto.LoginCheckResponse;
+import roomescape.controller.api.member.dto.MemberLoginRequest;
+import roomescape.controller.api.member.dto.TokenResponse;
+import roomescape.infrastructure.CookieExtractor;
+import roomescape.service.MemberService;
+
+@Controller
+@RequestMapping("/auth")
+public class LoginApiController {
+
+    private final MemberService memberService;
+    private final CookieExtractor cookieExtractor;
+
+    public LoginApiController(final MemberService memberService, final CookieExtractor cookieExtractor) {
+        this.memberService = memberService;
+        this.cookieExtractor = cookieExtractor;
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<TokenResponse> login(@RequestBody @Valid final MemberLoginRequest request,
+                                               final HttpServletResponse response) {
+        final String token = memberService.login(request);
+        final Cookie cookie = cookieExtractor.createCookie(token, 3600);
+        response.addCookie(cookie);
+        return ResponseEntity.ok().body(new TokenResponse(token));
+    }
+
+    @GetMapping("/login/check")
+    public ResponseEntity<LoginCheckResponse> checkLogin(final HttpServletRequest request) {
```

Comment:
HttpServletRequest 대신에 ArgumentResolver를 활용해볼 수도 있지 않을까요?

```suggestion
    public ResponseEntity<LoginCheckResponse> checkLogin(@LoginMember final Member member) {
```

Reply:
제안해주신 것처럼, `HttpServletRequest를` 직접 사용하는 대신 `ArgumentResolver`를 활용해 로그인 정보를 주입하는 방식이 훨씬 더 깔끔하고 객체지향적인 접근이라고 생각합니다. `@LoginMember`와 `HandlerMethodArgumentResolver`를 활용하면 인증 관련 로직을 한 곳에서 집중적으로 관리할 수 있고, 컨트롤러는 유즈케이스 처리에만 집중할 수 있어 책임 분리가 명확해집니다.
말씀해주신 방향을 반영하여 개선하겠습니다.